#160019076 Add Wish-List Implementation

iverenshaguy opened the pull request:
#### What does this PR do?
This pull request adds the wish list implementation to the platform
#### Description of Task to be completed?
- customers can add products to their wish-list
#### How should this be manually tested?
N/A
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
#160019076
#### Screenshots or gifs (if appropriate):
<img width="1440" alt="screen shot 2018-08-28 at 3 16 57 pm" src="https://user-images.githubusercontent.com/22734663/44730049-faf74600-aad7-11e8-835a-da3f613a76f7.png">

#### Questions:
N/A

phemonick commented on imports/plugins/custom/product-variant/client/components/customProductGridItem.js:
```diff
@@ -2,6 +2,7 @@ import React from "react";
 import { FacebookShareButton, FacebookIcon, TwitterShareButton, TwitterIcon, WhatsappShareButton, WhatsappIcon } from "react-share";
 import ReactStars from "react-stars";
 import { formatPriceString } from "/client/api";
+import { Components } from "@reactioncommerce/reaction-components";
```
line 5 and 6 should be combined because they are imported from same package.

iverenshaguy replied:
Thank you for spotting this. I'll fix it. Any other thing?

iverenshaguy committed:
feat(wish-list): add wish-list implementation

- add wish-list implementation for customers
- fix CI test failure

[Finishes #160019076]

phemonick reviewed (approved):
Excellent work. Keep up the good work!

nouwatinjacob reviewed (approved):
Hi ivenrenshaguy,
You did a good job. Keep it up
